File: translated/gemini/program_9.py
def main():
    N = int(input("Enter a non-negative integer: "))

    factorial = 1  # Use int for potentially large factorials
    for i in range(1, N + 1):
        factorial *= i

    print("The factorial of " + str(N) + " is: " + str(factorial))

if __name__ == "__main__":
    main()

File: input_program/program_9.java
import java.util.Scanner;

public class FactorialCalculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a non-negative integer: ");
        int N = scanner.nextInt();

        long factorial = 1; // Use long for potentially large factorials
        for (int i = 1; i <= N; i++) {
            factorial *= i;
        }

        System.out.println("The factorial of " + N + " is: " + factorial);
        scanner.close();
    }
}